react/features/base/util/uri.js

```diff
@@ -96,6 +96,43 @@ function _fixURIStringScheme(uri: string) {
     return uri;
 }
 
+/**
+ * Converts a room name to a backend-safe format. Properly lowercased and url encoded.
+ *
+ * @param {string?} room - The room name to convert.
+ * @returns {string?}
+ */
+export function getBackendSafeRoomName(room: ?string): ?string {
+    if (!room) {
+        return room;
+    }
+
+    /* eslint-disable no-param-reassign */
+    try {
+        // We do not know if we get an already encoded string at this point
+        // as different platforms do it differently, but we need a decoded one
+        // for sure. However since decoding a non-encoded string is a noop, we're safe
+        // doing it here.
+        room = decodeURIComponent(room);
+    } catch (e) {
+        // This can happen though if we get an unencoded string and it contains
+        // some characters that look like an encoded entity, but it's not.
+        // But in this case we're fine goin on...
+    }
+
+    // Only decoded strings can be lowercased properly
```

Comment:
punctuation, end the sentence in a period.

Reply:
Also, I think we should elaborate a bit here: we use locale aware lower casing because xyz...